cmake/FetchDependencies.cmake

```diff
@@ -1,9 +1,24 @@
+function(disable_warnings_for_target target)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        target_compile_options(${target} PRIVATE -w)
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        target_compile_options(${target} PRIVATE /W0)
+    endif()
+endfunction()
+
+function(disable_warnings_for_headers target)
+    get_target_property(target_include_dirs ${target} INTERFACE_INCLUDE_DIRECTORIES)
+    if (target_include_dirs)
+        target_include_directories(${target} SYSTEM INTERFACE ${target_include_dirs})
+    endif()
+endfunction()
+
+include(FetchContent)
+
 #############################
```

Comment:
It feels a little weird to have this banner style comment not at the top of the file anymore. Maybe just make it a standard comment?